test/test_type_promotion.py

```diff
@@ -402,6 +403,51 @@ def _test_spot(a, b, res_dtype):
                    torch.tensor(1., dtype=torch.complex64, device=device), torch.complex128)
         _test_spot(torch.tensor([1, 1], dtype=torch.bool, device=device), 1., torch.get_default_dtype())
 
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    def test_result_type_variadic(self, device):
+        # Floating dtypes are not listed here because the `numpy.result_type` may behave differently
+        # if inputs of mixed dtypes are given. The tests for floating type inputs are left to
+        # `test_result_type` and `test_result_type_tensor_vs_scalar`.
+        dtypes = [
+            torch.bool, torch.uint8, torch.int8, torch.int16, torch.int64,
+            torch.float16, torch.float32, torch.float64,
+            torch.complex64, torch.complex128,
+        ]
+        inputs = [
+            True,  # bool scalar
+            10,  # int scalar
+            10.0,  # float scalar
+            10j,  # complex scalar
+            *[torch.tensor([1, 2, 3], dtype=dtype) for dtype in dtypes],  # tensors
```

Comment:
This test is always running on the CPU because it's ignoring the device arg.

Does the current implementation work with CUDA tensors? What if both CUDA and CPU tensors are passed?

Reply:
I added tests for CUDA/mixed device tensors and confirmed that the current implementation passes the tests!